config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -116,8 +121,8 @@
                     if (isOptionalProperty(clazz, beanProp)) {
                         continue;
                     }
-                    // Otherwise, raise a {@link Missing} exception right here
-                    throw new ConfigException.Missing(beanProp.getName());
+                    // Otherwise, raise a {@link BugOrBroken} exception
+                    throw new ConfigException.BugOrBroken("Should have detected missing props earlier");
                 }
                 Object unwrapped = getValue(clazz, parameterType, parameterClass, config, configPropName);
```

Comment:
To let this getValue return null, I'd suggest passing in a parameter to getValue `optional=true|false`, and then at the top of `getValue` do something like:

```
if (optional and config.getIsNull(configPropName))
  return null;
```

I think that will work ?